### PR TITLE
Contact Form: add values shortcode parameter set value for each option field

### PR DESCRIPTION
**Pull request by ysupr**
##This is enhancement for #5280

This will allow user to set values for fields that has options ( radio, multiple checkbox, dropdown ). To set value.

**step to reproduce**
1. create page
2. insert contact form
3. insert field radio, multiple checkbox, and dropdown
4. adding values attributes, separated with comma (see sample below)
5. save

**sample shortcode**
[contact-form]
[contact-field label='dropdown' type='select' options='dropdown 1, dropdown 2' **values='d1,d2'**/]
[contact-field label='Radio' type='radio' options='Radio 1,Radio 2' **values='r1,r2'**/]
[contact-field label='Checkbox' type='checkbox-multiple' options='Checkbox 1,Checkbox 2' **values='c1,c2'**/]
[/contact-form]

**some cases**
1. if no values provided, it will uses options label as value (like previous version)
2. if value not matching with options (like, 2 options but 1 value) then only first one will use custom value. the second option swill use option (like point no 1)

### CODE REVIEW

**Comment by ysupr** [comment]
just noticed the conflict, still fixing it

**Comment by ysupr** [comment]
cc @jeherve

**Comment by ysupr** [comment]
got error on travis-ci for PHP7

`No output has been received in the last 10 minutes, this potentially indicates a stalled build or something wrong with the build itself.`

and not sure how to ask travis ci to do re-test

File: modules/contact-form/grunion-contact-form.php [hunk 1]
```diff
@@ -2199,6 +2199,7 @@ function __construct( $attributes, $content = null, $form = null ) {
 			'options'     => array(),
 			'id'          => null,
 			'default'     => null,
+            'values'      => null,
```
Comment (jeherve) [inline modules/contact-form/grunion-contact-form.php]: There seems to be an indentation issue here.

File: modules/contact-form/grunion-contact-form.php [hunk 2]
```diff
@@ -2217,6 +2218,10 @@ function __construct( $attributes, $content = null, $form = null ) {
 		// parse out comma-separated options list (for selects, radios, and checkbox-multiples)
 		if ( !empty( $attributes['options'] ) && is_string( $attributes['options'] ) ) {
 			$attributes['options'] = array_map( 'trim', explode( ',', $attributes['options'] ) );
+
+            if( !empty( $attributes['values'] ) && is_string( $attributes['values'] ) ) {
```
Comment (jeherve) [inline modules/contact-form/grunion-contact-form.php]: Could you add the necessary spaces around brackets to respect WordPress' coding standards?
https://make.wordpress.org/core/handbook/best-practices/coding-standards/php/#space-usage

File: modules/contact-form/grunion-contact-form.php [hunk 3]
```diff
@@ -2321,6 +2326,22 @@ function validate() {
...
+     * Check the default value for options field
+     *
+     * @param string value
+     * @param int index
+     * @param string default value
+     *
+     * @return string
+     */
+    public function get_option_value( $value, $index, $options ) {
+        if( empty( $value[ $index ] ) ) {
```
Comment (jeherve) [inline modules/contact-form/grunion-contact-form.php]: Could you add a space after `if` for coding standards?

File: modules/contact-form/grunion-contact-form.php [hunk 4]
```diff
@@ -2447,9 +2468,9 @@ function render() {
 			$r .= "\n<div>\n";
 			$r .= "\t\t<label for='" . esc_attr( $field_id ) . "' class='grunion-field-label select" . ( $this->is_error() ? ' form-error' : '' ) . "'>" . esc_html( $field_label ) . ( $field_required ? '<span>'. $required_field_text . '</span>' : '' ) . "</label>\n";
 			$r .= "\t<select name='" . esc_attr( $field_id ) . "' id='" . esc_attr( $field_id ) . "' " . $field_class . ( $field_required ? "required aria-required='true'" : "" ) . ">\n";
-			foreach ( $this->get_attribute( 'options' ) as $option ) {
-				$option = Grunion_Contact_Form_Plugin::strip_tags( $option );
-				$r .= "\t\t<option" . selected( $option, $field_value, false ) . ">" . esc_html( $option ) . "</option>\n";
+			foreach ( $this->get_attribute( 'options' ) as $optionIndex => $option ) {
+                $option = Grunion_Contact_Form_Plugin::strip_tags( $option );
```
Comment (jeherve) [inline modules/contact-form/grunion-contact-form.php]: There seems to be an indentation issue here, as well as a trailing space.

**Comment by ysupr** [comment]
Ok, @jeherve i have pushed new commit regarding issue with the space and indentation.

**Comment by ysupr** [comment]
Hi @jeherve i updated again. Here what i did.
1. installing phpcs and WordPress code standard, something missed before
2. run `phpcs --standar=WordPress grunion-contact-form.php` and found xxx errors on various issue
3. run `phpcbf --standar-WordPress grunion-contact-form.php` to fix some issue
4. do manual check based on your latest feedback to check if phpcbf fix it correctly,
5. find `if(` , `foreach(` and etc to check  if found another missed,
6. run the contact form, test to fill it, to check there everything is good

**Comment by jeherve** [comment]
Thanks a lot!

> run phpcs --standar=WordPress grunion-contact-form.php and found > 2500 errors on various issue

That's indeed to be expected. [We typically do not refactor code just to fix coding standards](https://make.wordpress.org/core/handbook/contribute/code-refactoring/), so old code sometimes doesn't meet the standards. However, whenever we touch that code or introduce something new, we try to follow the standards as much as possible.

**Comment by dereksmart** [comment]
Needs a rebase please!

**Comment by ysupr** [comment]
hi @dereksmart this is my first rebasing, little surprised that it loaded many commit on this issue. is it normal?

**Comment by jeherve** [comment]
It's not, but you can fix it 👍 . It most likely happened because your copy of `master` wasn't up to date with the main repository's `master`.

To fix it, you can start by updating your version of `master` with all the changes from the main repo's `master`. Then, you can follow the instructions here to perform a new rebase:
https://github.com/edx/edx-platform/wiki/How-to-Rebase-a-Pull-Request

Let us know how it goes!

**Comment by ysupr** [comment]
Ok, thanks @jeherve, hope it's ok now. what do you think @dereksmart ?

**Comment by jeherve** [comment]
@ysupr I'm afraid it's still not correct, as the PR still lists a lot of commits that are already in the `master` branch. Could you give it another try?

**Comment by ysupr** [comment]
ah ok, i will try it again. so when it successed, all other commit should be not showing here, right?

**Comment by jeherve** [comment]
Correct! The end result should look like this:
https://github.com/jeherve/jetpack/tree/add/set-value-for-options-fields

If you're having trouble fixing this, let me know and we'll work from the branch above instead.

**Comment by ysupr** [comment]
ok, finally finish the rebasing, thanks for your input @jeherve 

cc @dereksmart

File: modules/contact-form/grunion-contact-form.php [hunk 5]
```diff
@@ -176,14 +177,15 @@ function __construct() {
 			add_action( 'template_redirect', array( $this, 'process_form_submission' ) );
 		}
 
-		/* Can be dequeued by placing the following in wp-content/themes/yourtheme/functions.php
+		/*
+		 Can be dequeued by placing the following in wp-content/themes/yourtheme/functions.php
```
Comment (ebinnion) [inline modules/contact-form/grunion-contact-form.php]: We need a `*` before this line.

File: modules/contact-form/grunion-contact-form.php [hunk 6]
```diff
@@ -1046,26 +1052,27 @@ protected static function make_csv_row_from_feedback( $post_id, $fields ) {
...
-			if ( $field === __( 'Contact Form', 'jetpack' ) ) // the first field will ever be the contact form, so we can continue
+			if ( $field === __( 'Contact Form', 'jetpack' ) ) { // the first field will ever be the contact form, so we can continue
 				continue;
-			elseif ( array_key_exists( $field, $all_fields ) )
-				$row_items[] = $all_fields[$field];
-			else
-				$row_items[] = '';
+			} elseif ( array_key_exists( $field, $all_fields ) ) {
+				$row_items[] = $all_fields[ $field ];
+			} else { $row_items[] = '';
```
Comment (ebinnion) [inline modules/contact-form/grunion-contact-form.php]: Formatting is off here.

**Comment by ysupr** [comment]
Hi @ebinnion sorry for really noob question, since this PR are merged, what should i do with your code feedback? should i make new PR? or i can just push it here?

**Comment by jeherve** [comment]
@ysupr You'll need to create a new PR, with a new branch.

**Pull request merged**